File: front/mini-app/src/api/album.ts
import { request } from '@/utils/request'
import { getStoredUserId } from '@/utils/auth'

const DEFAULT_API_BASE_URL = import.meta.env.VITE_API_BASE_URL || 'https://wikiroco.com'
const API_BASE_URL = DEFAULT_API_BASE_URL.replace(/\/+$/, '')

export interface AlbumPhoto {
  id: number
  user_id: number
  pet_id: number
  image_url: string
  is_featured: boolean
  created_at: string | null
  updated_at: string | null
}

export interface AlbumCreatePayload {
  user_id: number
  pet_id: number
  image_url: string
  is_featured?: boolean
}

/** 查询某用户某宠物的相册（精选优先，后端已排序）。 */
export function fetchAlbumPhotos(userId: number | string, petId: number) {
  return request<AlbumPhoto[]>({
    url: '/api/album',
    data: { user_id: userId, pet_id: petId },
  })
}

/** 新增一张相册照片。 */
export function createAlbumPhoto(payload: AlbumCreatePayload) {
  return request<AlbumPhoto>({
    url: '/api/album',
    method: 'POST',
    data: payload,
  })
}

/** 设置 / 取消精选。 */
export function setAlbumPhotoFeatured(photoId: number, isFeatured: boolean) {
  return request<AlbumPhoto>({
    url: `/api/album/${photoId}/featured`,
    method: 'POST',
    data: { is_featured: isFeatured },
  })
}

/** 删除一张照片（后端会同步删除 OSS 对象）。 */
export function deleteAlbumPhoto(photoId: number) {
  return request<AlbumPhoto>({
    url: `/api/album/${photoId}`,
    method: 'DELETE',
  })
}

/** 上传本地图片到 OSS，返回访问 URL。走 uni.uploadFile（multipart）。 */
export function uploadAlbumImage(filePath: string): Promise<string> {
  const userId = getStoredUserId()
  return new Promise((resolve, reject) => {
    uni.uploadFile({
      url: `${API_BASE_URL}/api/file/upload`,
      filePath,
      name: 'file',
      formData: { prefix: 'album' },
      header: userId ? { Authorization: userId } : undefined,
      success: (res) => {
        if (res.statusCode >= 200 && res.statusCode < 300) {
          try {
            const body = JSON.parse(res.data) as { url?: string }
            if (body.url) {
              resolve(body.url)
              return
            }
            reject(new Error('上传响应缺少 url'))
          } catch {
            reject(new Error('上传响应解析失败'))
          }
          return
        }
        reject(new Error(`上传失败（${res.statusCode}）`))
      },
      fail: (err) => reject(new Error(err.errMsg || '上传失败')),
    })
  })
}

File: api/schemas/album.py
from datetime import datetime

from pydantic import BaseModel, Field


class AlbumPhoto(BaseModel):
    """一张相册照片。"""

    id: int
    user_id: int
    pet_id: int
    image_url: str
    is_featured: bool = False
    created_at: datetime | None = None
    updated_at: datetime | None = None


class AlbumCreateRequest(BaseModel):
    """新增相册照片：用户 id + 宠物 id + 图片 URL。"""

    user_id: int = Field(..., description="用户 id")
    pet_id: int = Field(..., description="宠物（精灵）id")
    image_url: str = Field(..., min_length=1, description="图片访问 URL（OSS 地址）")
    is_featured: bool = Field(False, description="是否直接标记为精选")


class AlbumFeaturedRequest(BaseModel):
    """设置 / 取消某张照片的精选状态。"""

    is_featured: bool = Field(..., description="True 设为精选，False 取消精选")
